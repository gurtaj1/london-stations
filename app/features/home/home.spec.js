import React from 'react';
import { shallow } from 'enzyme';

import Home from 'features/home/home';
import fixture from 'features/home/home.fixture';

describe('features.home', () => {
  describe('Home', () => {
    const props = fixture;

    afterEach(() => {
      jest.clearAllMocks();
    });

    it('should export something', () => {
      expect(Home).toBeDefined();
    });

    it('should render without crashing', () => {
      const wrapper = shallow(<Home {...props} />);

      expect(wrapper).toMatchSnapshot();
    });

    it('should call props.history.push if props.error is truthy', () => {
      const updatedProps = {
        ...props,
        home: {
          ...props.home,
          homeData: null,
          error: {}
        }
      };
      const wrapper = shallow(<Home {...props} />);
      wrapper.setProps(updatedProps);

      expect(updatedProps.history.push).toHaveBeenCalled();
    });

    it('should not call props.history.push if props.error is falsy', () => {
      const updatedProps = {
        ...props,
        home: {
          ...props.home,
          homeData: [
            {
              name: 'station-name-2',
              tubeLines: [
                'central'
              ]
            }
          ]
        }
      };
      const wrapper = shallow(<Home {...props} />);
      wrapper.setProps(updatedProps);

      expect(updatedProps.history.push).not.toHaveBeenCalled();
    });
  });
});
