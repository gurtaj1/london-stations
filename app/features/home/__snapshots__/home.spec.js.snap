// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features.home Home should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    history={
      Object {
        "push": [MockFunction],
      }
    }
    home={
      Object {
        "homeData": Array [
          Object {
            "name": "station-name",
            "tubeLines": Array [
              "circle",
              "district",
            ],
          },
        ],
        "isFetching": false,
      }
    }
    requestHomePage={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Station
          name="station-name"
          tubeLines={
            Array [
              "circle",
              "district",
            ]
          }
        />,
      ],
      "className": "home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "station-name",
        "nodeType": "function",
        "props": Object {
          "name": "station-name",
          "tubeLines": Array [
            "circle",
            "district",
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Station
            name="station-name"
            tubeLines={
              Array [
                "circle",
                "district",
              ]
            }
          />,
        ],
        "className": "home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "station-name",
          "nodeType": "function",
          "props": Object {
            "name": "station-name",
            "tubeLines": Array [
              "circle",
              "district",
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
